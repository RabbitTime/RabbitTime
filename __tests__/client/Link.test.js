import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';


it('changes the class when hovered', () => {
  const component = renderer.create(
    <Link page="http://www.test.com">Test</Link>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  // trigger the callback
  renderer.act(() => {
    tree.props.onMouseEnter();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  // trigger the callback
  renderer.act(() => {
    tree.props.onMouseLeave();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});