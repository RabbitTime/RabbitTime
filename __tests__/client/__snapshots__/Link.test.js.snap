// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://www.test.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</a>
`;

exports[`changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.test.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</a>
`;

exports[`changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://www.test.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test
</a>
`;
